templates/connect-inject-deployment.yaml

```diff
@@ -80,8 +81,8 @@ spec:
               consul-k8s inject-connect \
                 -default-inject={{ .Values.connectInject.default }} \
                 -consul-image="{{ default .Values.global.image .Values.connectInject.imageConsul }}" \
-                {{ if .Values.connectInject.imageEnvoy -}}
-                -envoy-image="{{ .Values.connectInject.imageEnvoy }}" \
+                {{ if .Values.global.imageEnvoy -}}
```

Comment:
should we always require it? that way we can eventually get rid of the default in consul-k8s?

Reply:
my plan was to take up the consul-k8s work as well since it's fairly easy and raise a PR there to make it mandatory and remove the defaults.
I suppose it makes sense to hold off on merging this until that work is done?

Reply:
Yeah let's not support `global.imageEnvoy: null` since from the beginning this value has had a default.

Re consul-k8s, I don't think you need to wait on merging this PR for that consul-k8s change.

Reply:
Cool, sounds good. Yeah, I agree, no need to block this PR on the consul-k8s work. 

If we require it, then we should remove this `if` condition. Having the `if` means that you can still set it to `null` in the helm chart and use consul-k8s default.